syntax = "proto3";

package network;

import "google/protobuf/timestamp.proto";

// Wi-Fi interface metrics
message WifiMetrics {
  string node_id = 1;
  string interface_name = 2;
  google.protobuf.Timestamp timestamp = 3;
  
  repeated string flags = 4;
  uint32 mtu = 5;
  string ipv4 = 6;
  string netmask = 7;
  string broadcast = 8;
  repeated string ipv6 = 9;
  string mac = 10;
  
  uint64 rx_packets = 11;
  uint64 rx_bytes = 12;
  uint64 tx_packets = 13;
  uint64 tx_bytes = 14;
  uint32 rx_errors = 15;
  uint32 tx_errors = 16;
}

// Acknowledgment from server
message MetricsAck {
  string node_id = 1;
  bool success = 2;
  string message = 3;
  google.protobuf.Timestamp server_timestamp = 4;
}

// Service definition
service NetworkMonitoring {
  // Bidirectional streaming: client pushes Wi-Fi metrics, server sends acks
  rpc StreamWifiMetrics(stream WifiMetrics) returns (stream MetricsAck);
}